src/main.rs

```
@@ -266,6 +285,28 @@ where
     Ok(())
 }
 
+fn check_log(log: OffsetLog<u32>) -> Result<(), Error> {
+    let stdin = stdin();
+    let mut stdout = stdout().into_raw_mode()?;
+
+    println!("total entries: {}",log.end());
+
+    // TODO: option for reverse?
+    log.iter().for_each(|e| {
+        let v: Result<Value, serde_json::error::Error> = serde_json::from_slice(&e.data);
+
+        match v {
+            Ok(v) => print!("\rentry {} ok", e.offset),
+            Err(err) => { 
+                println!("\n\n===>found broken entry: {} (err: {})", e.offset, err);
+                return;
```

Should we `return` here? I mean, we could just find all the errors right?

---

If just want to find the first error you can do something like this:
```rust

    let result = log.iter()
        .map(|e| {
            serde_json::from_slice::<Value>(&e.data)
                .map(|_|())
                .map_err(|_| e)
        })
        .collect::<Result<(), LogEntry>>();

    if let Err(e) = result {
        println!("\n\n===>found broken entry: {}", e.offset);
    }
```

Which is super crazy if you think about it. It's collecting all those entries down into one `Result`. So if they were all ok, the result will be `Ok(())` but if one fails, it fails fast and then you'll have an `Err(LogEntry)`

The `if let` syntax is nice when you don't need all the arms of a `match`

---

>  I mean, we could just find all the errors right?

Yea, good question. I really just wanted to see if there is any error but I see your point.

General question about this chunk: Does the last line of a block return it's values automatically without a `return` statement?

```rust
        .map(|e| {
            serde_json::from_slice::<Value>(&e.data)
                .map(|_|())
                .map_err(|_| e)
        })
```

---

> Does the last line of a block return it's values automatically without a return statement?

Yeah! It's nice, if you omit the semicolon on the last line it's an implicit return.